pyproject.toml

```diff
@@ -2,7 +2,8 @@
 build-backend = "setuptools.build_meta"
 requires = [
     "certifi>=2020.06.20",
-    "oldest-supported-numpy",
+    "oldest-supported-numpy; platform_python_implementation!='PyPy'",
+    "numpy>=1.25; platform_python_implementation=='PyPy'",
```

Comment:
My current understanding is that we _should_ actually just be able to do numpy>=1.25 independent of platform since numpy/numpy#23528

Reply:
Is our set of supported Pythons a subset of the versions that numpy supports in 1.25?

Reply:
answering my own question: yes

I'll do that too.

Reply:
It defaults to going to 1.17, which is perhaps a bit _too_ far back... but also not _really_ a harm in that, unless we actually know we need something newer? (At which point adding a `#define` can change the version, just not sure where the most sensible place for that define is)

Reply:
Specifically, it allows you to put in the C code a line like 

```c
#define NPY_TARGET_VERSION NPY_1_22_API_VERSION
```

To control the python version. the default is `NPY_1_17_API_VERSION`

If we use features more recent (and the macro is not defined by e.g. pybind11/cibuildwheel already), we may wish to put the line specify 1.22 somewhere.

Reply:
We should do that in a follow-up PR.

If it defaults to an API version older than we actually support then we are sure it will work :)